addons/l10n_id_efaktur/models/efaktur.py

```diff
@@ -14,44 +14,55 @@ class Efaktur(models.Model):
     company_id = fields.Many2one('res.company', required=True, default=lambda self: self.env.company)
     max = fields.Char(compute='_compute_default', store=True, readonly=False)
     min = fields.Char(compute='_compute_default', store=True, readonly=False)
-    available = fields.Integer(compute='_compute_available', store=True)
+    available = fields.Integer(readonly=True)
+    next_num = fields.Integer(compute="_compute_next_num", store=True)
+
+    @api.depends('max', 'available')
+    def _compute_next_num(self):
+        for record in self:
+            record.next_num = int(record.max) - record.available + 1
 
     @api.model
     def pop_number(self, company_id):
-        range = self.search([('company_id', '=', company_id)], order="min ASC", limit=1)
-        if not range:
+        erange = self.search([('company_id', '=', company_id)], order="next_num ASC", limit=1)
```

Comment:
I think you only want the ranges that have available numbers no? and then you can keep the order as before